src/sql/services/bootstrap/bootstrapService.ts

```diff
@@ -68,63 +30,20 @@ function createUniqueSelector(selector: string): string {
-			{ provide: IRestoreService, useValue: collection.get(IRestoreService) },
-			{ provide: IBackupService, useValue: collection.get(IBackupService) },
-			{ provide: IBackupUiService, useValue: collection.get(IBackupUiService) },
-			{ provide: IAngularEventingService, useValue: collection.get(IAngularEventingService) },
-			{ provide: IInsightsDialogService, useValue: collection.get(IInsightsDialogService) },
-			{ provide: ISqlOAuthService, useValue: collection.get(ISqlOAuthService) },
-			{ provide: IFileBrowserService, useValue: collection.get(IFileBrowserService) },
-			{ provide: IFileBrowserDialogController, useValue: collection.get(IFileBrowserDialogController) },
-			{ provide: IClipboardService, useValue: collection.get(IClipboardService) },
-			{ provide: ICapabilitiesService, useValue: collection.get(ICapabilitiesService) },
-			{ provide: IDashboardViewService, useValue: collection.get(IDashboardViewService) },
-			{ provide: IModelViewService, useValue: collection.get(IModelViewService) },
 
-			// vscode services
-			{ provide: vsIClipboardService, useValue: collection.get(vsIClipboardService) },
-			{ provide: IKeybindingService, useValue: collection.get(IKeybindingService) },
-			{ provide: IContextKeyService, useValue: collection.get(IContextKeyService) },
-			{ provide: IContextMenuService, useValue: collection.get(IContextMenuService) },
-			{ provide: IContextViewService, useValue: collection.get(IContextViewService) },
-			{ provide: IWorkbenchEditorService, useValue: collection.get(IWorkbenchEditorService) },
-			{ provide: IPartService, useValue: collection.get(IPartService) },
-			{ provide: IInstantiationService, useValue: collection.get(IInstantiationService) },
-			{ provide: IConfigurationService, useValue: collection.get(IConfigurationService) },
-			{ provide: IWorkspaceContextService, useValue: collection.get(IWorkspaceContextService) },
-			{ provide: IAccountManagementService, useValue: collection.get(IAccountManagementService) },
-			{ provide: IWindowsService, useValue: collection.get(IWindowsService) },
-			{ provide: IWindowService, useValue: collection.get(IWindowService) },
-			{ provide: ITelemetryService, useValue: collection.get(ITelemetryService) },
-			{ provide: IStorageService, useValue: collection.get(IStorageService) },
-			{ provide: ICommandService, useValue: collection.get(ICommandService) },
-			{ provide: IJobManagementService, useValue: collection.get(IJobManagementService) },
-			{ provide: IEnvironmentService, useValue: collection.get(IEnvironmentService) },
-			{ provide: INotificationService, useValue: collection.get(INotificationService) },
-			{ provide: IWorkbenchThemeService, useValue: collection.get(IWorkbenchThemeService) },
-			{ provide: IProgressService, useValue: collection.get(IProgressService) }
-		];
+		const providers: Provider = [];
+
+		_util.serviceIds.forEach(id => {
+			providers.push({ provide: id, useFactory: () => (<any>service)._getOrCreateServiceInstance(id) });
```

Comment:
Would it be better to make `_getOrCreateServiceInstance` public instead of calling it like this?